src/Mod/TechDraw/Gui/QGIViewDimension.cpp

```diff
@@ -112,7 +112,7 @@ QGIDatumLabel::QGIDatumLabel() : m_dragState(NoDrag)
     m_tolTextUnder->setParentItem(m_textItems);
     m_unitText = new QGCustomText();
     m_unitText->setTightBounding(true);
-    m_unitText->setParentItem(m_textItems);
+    m_unitText->setParentItem(nullptr);
```

Comment:
I think it is null by default so `setParentItem` shouldn't be needed.
Or are you explicitly setting it here to show the reader that it is not just missed? (A comment it shouldn't be set to `m_textItems` would be good.

Btw, why is that? Will an empty unit still have a size that increases the box?

Reply:
Mostly, even empty it's bounding box has width and height, but also in the case where it's not being used its x positions increases the bounding box  size calculation.

Reply:
Removed code and added comment.